examples/sha256/table16/compression/compression_gates.rs

```diff
@@ -56,14 +63,12 @@ impl<F: FieldExt> CompressionGate<F> {
             + spread_word_lo * (-F::one())
             + spread_word_hi * F::from_u64(1 << 32) * (-F::one());
 
-        CompressionGate(
-            s_decompose_abcd
-                * (range_check_tag_b
-                    + range_check_tag_d
-                    + dense_check
-                    + spread_check
-                    + check_spread_and_range),
-        )
+        check_spread_and_range
+            .chain(Some(("range_check_tag_b", range_check_tag_b)))
+            .chain(Some(("range_check_tag_d", range_check_tag_d)))
+            .chain(Some(("dense_check", dense_check)))
+            .chain(Some(("spread_check", spread_check)))
+            .map(move |(name, poly)| (name, s_decompose_abcd.clone() * poly))
```

Comment:
Were the gates using the `selector * (check1 + check2 + ...)` pattern previously incorrect?

Reply:
Yes, the checks should have been linearly independent instead of being summed.